src/anndata/_core/views.py

```diff
@@ -474,6 +582,20 @@ def _resolve_idx_slice(
         return np.arange(*old.indices(l))[new]
 
 
+@_resolve_idx.register(jnp.ndarray)
```

Comment:
`_resolve_idx` has no default case.  I would move 

1. change `_resolve_idx_jnp` to `_resolve_idx_array_api` and make it generic over the array api.  If this function is tested right now via `jax` it should still be testable in and work when you use the array-api to do these operations instead of `jnp`
2. Remove `@_resolve_idx.register(jnp.ndarray)` and then immediately
3. Move the body of this function into the default implementation i.e., `_resolve_idx` with the `@singledispatch` above it.  This will allow you to _by default_ check if something is array-api compatible and then use your implementation. 
4. Otherwise just raise the `NotImplementedError` that is already there